### PR TITLE
Small refactoring of methods/mod.rs

### PR DESCRIPTION
**Pull request by jeremystucki**
changelog: none

### CODE REVIEW

**Comment by phansch** [comment]
Thanks for tackling `methods/mod.rs`!

@bors r+

**Comment by phansch** [comment]
I took the liberty to add the `changelog: none` the the PR body, now it should work.

@bors retry

File: clippy_lints/src/methods/mod.rs [hunk 1]
```diff
@@ -1289,46 +1281,37 @@ fn lint_or_fun_call<'a, 'tcx>(
...
+        if_chain! {
+            if know_types.iter().any(|k| k.2.contains(&name));
 
-        let mut finder = FunCallFinder { cx: &cx, found: false };
-        finder.visit_expr(&arg);
-        if !finder.found {
-            return;
-        }
+            let mut finder = FunCallFinder { cx: &cx, found: false };
+            if { finder.visit_expr(&arg); finder.found };
```
Comment (mikerite) [inline clippy_lints/src/methods/mod.rs]: This should have been flagged by `block_in_if_condition_stmt`

Comment (phansch) [reply clippy_lints/src/methods/mod.rs]: This may have been skipped because of a macro check in the lint?

**Pull request merged**